Lombiq.Hosting.Azure.ApplicationInsights/ApplicationInsightsOptions.cs

```diff
@@ -71,12 +71,26 @@ public class ApplicationInsightsOptions
     public bool EnableClientSideTracking { get; set; } = true;
 
     /// <summary>
-    /// Gets or sets a value indicating whether local development is enabled where telemetry is sent to
-    /// Azure Application Insights from the local instance.
-    /// Will only work if local authentication is enabled in the properties of the Applications Insights resource
-    /// which should only be the case during development and testing.
+    /// Gets or sets a value indicating whether local development is enabled where telemetry is sent to a secure Azure
+    /// Application Insights resource from the local instance. When using this option you will also need to provide
+    /// tenantId, clientId, and clientSecret of the service principal.
     /// </summary>
-    public bool EnableLocalDevelopment { get; set; }
+    public bool EnableSecureLocalDevelopment { get; set; }
+
+    /// <summary>
+    /// Gets or sets the tenant ID of the Azure AD application that's used for local development.
+    /// </summary>
+    public string TenantId { get; set; }
+
+    /// <summary>
+    /// Gets or sets the client ID of the Azure AD application that's used for local development.
+    /// </summary>
+    public string ClientId { get; set; }
+
+    /// <summary>
+    /// Gets or sets the client secret of the Azure AD application that's used for local development.
+    /// </summary>
+    public string ClientSecret { get; set; }
```

Comment:
These three configs are just for local auth of the control channel. So, not to pollute this type, it'd be better to put them into a separate class, like `SecureControlChannelLocalAuthenticationCredentials` and have a property with that, named the same.